client/src/setupProxy.js

```diff
@@ -0,0 +1,9 @@
+const { createProxyMiddleware } = require("http-proxy-middleware");
+module.exports = function (app) {
+	app.use(
+		["/users/me", "/users/logout"],
+		createProxyMiddleware({
+			target: "http://localhost:3001",
+		}),
+	);
+};
```

Comment:
Is this necessary? Package.json already has the proxy set up :
 ```js
"proxy": "http://localhost:3001"
```

Reply:
I removed the `http-proxy-middleware` and I configured the package.json like this:
`"proxy": {
    "/api/users/logout": {
        "target": "http://localhost:3001"
    }
}`

and I got this error:
```
**_When specified, "proxy" in package.json must be a string.
Instead, the type of "proxy" was "object".
Either remove "proxy" from package.json, or make it a string._**
```
And I read about it again here:
[https://stackoverflow.com/questions/52605997/when-specified-proxy-in-package-json-must-be-a-string](url)

So I went ahead and reinstalled the middleware.

Reply:
Yes, but that's because it does take a string, not an object. You need to keep it as 
`"proxy": "http://localhost:3001"`
This will proxy any POST requests or GET requests for which there's no frontend route. 

I also noticed from your backend that the route for logout is `/users/logout` rather than `/api/users/logout`

Reply:
Ok. I tested it before changing, and I think it didn't work as expected. But maybe I made a mistake. Anyways, I will change it back.
And about the route, I just checked and it was `/api/users/logout` . Maybe what you saw was in an older version before being merged to `dev`?